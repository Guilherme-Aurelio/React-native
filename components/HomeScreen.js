import React, { useState } from 'react';
import { View, Text, TouchableOpacity, Image, ImageBackground, StyleSheet } from 'react-native';

const HomeScreen = ({ navigation }) => {
    return (
      <ImageBackground 
        source={require('../assets/objects.png')} 
        style={styles.container}
        resizeMode="cover"
      >
        <View style={styles.contentContainer}>
          <View style={styles.header}>
            <Image style={styles.logo} source={require('../assets/logomarca.png')} />
          </View>
          <View style={styles.logoContainer}></View>
          <Text style={styles.headerText}>Bem vindos ao SEGUE-ME</Text>
          <Text style={styles.welcomeText}>Um Batalhão de Amor</Text>
          <View style={styles.containerButton}>
            <TouchableOpacity style={styles.button} onPress={() => navigation.navigate('Login')}>
              <Text style={styles.textButton}>Entrar</Text>
            </TouchableOpacity>
            <TouchableOpacity style={styles.button2} onPress={() => navigation.navigate('Registration')}>
              <Text style={styles.textButton2}>Cadastrar</Text>
            </TouchableOpacity>
          </View>
        </View>
      </ImageBackground>
    );
  };


const styles = StyleSheet.create({
    container: {
      flex: 1,
      width: '100%',
      height: '100%',
    },
    contentContainer: {
      flex: 1,
      justifyContent: 'center', // Centraliza verticalmente
      alignItems: 'center', // Centraliza horizontalmente
      padding: 20,
      position: 'relative', // Adicionado para posicionar o botão admin
    },
    header: {
      alignItems: 'center',
      marginBottom: 20,
    },
    logoContainer: {
      alignItems: 'center',
      marginBottom: 20,
    },
    headerText: {
      fontSize: 35,
      fontWeight: 'bold',
      textAlign: 'center',
      color: '#FFA500',
      marginBottom: 10,
      marginHorizontal: 50
    },
    welcomeText: {
      fontSize: 20,
      textAlign: 'center',
      marginBottom: 30,
      fontWeight: 'bold',
      paddingBottom: 40
    },
    button: {
      alignItems: 'center',
      backgroundColor: '#F9AF0B',
      padding: 15,
      marginHorizontal: 15,
      borderRadius: 5,
      shadowColor: '#000',
      shadowOffset: { width: 0, height: 2 },
      shadowOpacity: 0.3,
      shadowRadius: 8,
      elevation: 5,
      height: 50,
      width: 150
    },
    button2: {
      alignItems: 'center',
      backgroundColor: '#ffffff',
      padding: 15,
      marginHorizontal: 10,
      borderRadius: 5,
      shadowColor: '#000',
      shadowOffset: { width: 0, height: 2 },
      shadowOpacity: 0.3,
      shadowRadius: 8,
      elevation: 5,
      height: 50,
      width: 150
    },
    textButton: {
      fontWeight: 'bold',
      color: '#FFFFFF',
      fontSize: 16,
    },
    textButton2: {
      fontWeight: 'bold',
      color: '#000000',
      fontSize: 16,
    },
    logo: {
      width: 300,
      height: 300,
    },
    containerButton: {
      flexDirection: 'row',
      justifyContent: 'center',
    }
  });
  
  
  export default HomeScreen;
  
