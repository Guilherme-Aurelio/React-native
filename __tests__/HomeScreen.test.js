import React from 'react';
import { render, fireEvent } from '@testing-library/react-native';
import HomeScreen from '../components/HomeScreen'; // Substitua pelo caminho correto do seu componente

describe('HomeScreen', () => {
  const navigation = {
    navigate: jest.fn(), // Mock da função de navegação
  };

  it('deve renderizar os botões e permitir navegação', () => {
    const { getByTestId } = render(<HomeScreen navigation={navigation} />);

    // Verificar se o botão de Admin está presente
    const adminButton = getByTestId('admin-button');
    expect(adminButton).toBeTruthy();

    // Verificar se o botão de Login está presente
    const loginButton = getByTestId('login-button');
    expect(loginButton).toBeTruthy();

    // Verificar se o botão de Cadastro está presente
    const registerButton = getByTestId('register-button');
    expect(registerButton).toBeTruthy();

    // Simular clique no botão de Admin e verificar navegação
    fireEvent.press(adminButton);
    expect(navigation.navigate).toHaveBeenCalledWith('Admin');

    // Simular clique no botão de Login e verificar navegação
    fireEvent.press(loginButton);
    expect(navigation.navigate).toHaveBeenCalledWith('Login');

    // Simular clique no botão de Cadastro e verificar navegação
    fireEvent.press(registerButton);
    expect(navigation.navigate).toHaveBeenCalledWith('Registration');
  });
});

