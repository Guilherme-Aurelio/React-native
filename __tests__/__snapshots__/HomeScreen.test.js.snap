// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders HomeScreen correctly 1`] = `
<View
  accessibilityIgnoresInvertColors={true}
  style={
    {
      "backgroundColor": "#FFFFFF",
      "flex": 1,
      "height": "100%",
      "width": "100%",
    }
  }
>
  <Image
    resizeMode="cover"
    source={
      {
        "testUri": "../../../assets/objects.png",
      }
    }
    style={
      [
        {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        {
          "height": "100%",
          "width": "100%",
        },
        undefined,
      ]
    }
  />
  <View
    style={
      {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
        "padding": 20,
        "position": "relative",
      }
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
    >
      <Text>
        Admin
      </Text>
    </View>
    <View
      style={
        {
          "alignItems": "center",
          "marginBottom": 20,
        }
      }
    >
      <Image
        source={
          {
            "testUri": "../../../assets/logomarca.png",
          }
        }
        style={
          {
            "height": 300,
            "width": 300,
          }
        }
      />
    </View>
    <View
      style={
        {
          "alignItems": "center",
          "marginBottom": 20,
        }
      }
    />
    <Text
      style={
        {
          "color": "#FFA500",
          "fontSize": 35,
          "fontWeight": "bold",
          "marginBottom": 10,
          "marginHorizontal": 50,
          "textAlign": "center",
        }
      }
    >
      Bem vindos ao SEGUE-ME
    </Text>
    <Text
      style={
        {
          "fontSize": 20,
          "fontWeight": "bold",
          "marginBottom": 30,
          "paddingBottom": 40,
          "textAlign": "center",
        }
      }
    >
      Um Batalh√£o de Amor
    </Text>
    <View
      style={
        {
          "flexDirection": "row",
          "justifyContent": "center",
        }
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#F9AF0B",
            "borderRadius": 5,
            "elevation": 5,
            "height": 50,
            "marginHorizontal": 15,
            "opacity": 1,
            "padding": 15,
            "shadowColor": "#000",
            "shadowOffset": {
              "height": 2,
              "width": 0,
            },
            "shadowOpacity": 0.3,
            "shadowRadius": 8,
            "width": 150,
          }
        }
      >
        <Text
          style={
            {
              "color": "#FFFFFF",
              "fontSize": 16,
              "fontWeight": "bold",
            }
          }
        >
          Entrar
        </Text>
      </View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#ffffff",
            "borderRadius": 5,
            "elevation": 5,
            "height": 50,
            "marginHorizontal": 10,
            "opacity": 1,
            "padding": 15,
            "shadowColor": "#000",
            "shadowOffset": {
              "height": 2,
              "width": 0,
            },
            "shadowOpacity": 0.3,
            "shadowRadius": 8,
            "width": 150,
          }
        }
      >
        <Text
          style={
            {
              "color": "#000000",
              "fontSize": 16,
              "fontWeight": "bold",
            }
          }
        >
          Cadastrar
        </Text>
      </View>
    </View>
  </View>
</View>
`;
